app/controllers/sessions_controller.rb

```diff
@@ -19,7 +21,14 @@ def create
       options = { alert: t('flash.duplicate_nick', name: e.nickname) }
     end
 
-    redirect_to request.env['omniauth.origin'].presence || root_path, options
+    redirect_path = request.env['omniauth.origin'].presence || root_path
+
+    if current_user&.missing_name?
```

Comment:
I get the idea here, but i'm not a fan of unnecessary complexity in the signup process. it's complex as it is and users get distracted easily. i would recommend handling that in the view layer, maybe where the flash is rendered and hint to the edit page if the username has not yet been changed.